hack/release-from/action.go

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v3"
+	"gopkg.in/yaml.v2"
+)
+
+type component struct {
+	Image    string `yaml:"image"`
+	Version  string `yaml:"version"`
+	Registry string `yaml:"registry"`
+}
+
+type productConfig struct {
+	Title      string               `yaml:"title"`
+	Components map[string]component `yaml:"components"`
+}
+
+func releaseFrom(ctx context.Context, c *cli.Command) error {
+	// get root directory of operator git repo
+	repoRootDir, err := runCommand("git", []string{"rev-parse", "--show-toplevel"}, nil)
+	if err != nil {
+		return fmt.Errorf("Error getting git root directory: %s", err)
+	}
+
+	// fetch config from the base version of the operator
+	if err := retrieveBaseVersionConfig(baseOperatorFlag.Name, repoRootDir); err != nil {
+		return fmt.Errorf("Error getting base version config: %s", err)
+	}
+
+	// Apply new version overrides
+	calicoOverrides := c.StringSlice(exceptCalicoFlag.Name)
+	if len(calicoOverrides) > 0 {
+		if err := updateConfig(repoRootDir, calicoConfig, calicoOverrides); err != nil {
+			return fmt.Errorf("Error overriding calico config: %s", err)
+		}
+	}
+	enterpriseOverrides := c.StringSlice(exceptEnterpriseFlag.Name)
+	if len(enterpriseOverrides) > 0 {
+		if err := updateConfig(repoRootDir, enterpriseConfig, enterpriseOverrides); err != nil {
+			return fmt.Errorf("Error overriding calico config: %s", err)
+		}
+	}
+
+	// Either build a new release or a new hashrelease operator
+	version := c.String(versionFlag.Name)
+	release, err := isRelease(version)
+	if err != nil {
+		return fmt.Errorf("Error determining if version is a release: %s", err)
+	} else if release {
+		return newOperator(repoRootDir, version, c.String(remoteFlag.Name))
+	}
+
+	return newHashreleaseOperator(repoRootDir, version, c.StringSlice(archFlag.Name))
+}
+
+// isRelease checks if the version is a release version.
+// A release version is in the format vX.Y.Z.
```

Comment:
Useful comment, but could probably be more succinct:

```suggestion
// isRelease checks if the version is in the format vX.Y.Z.
```